### DISCUSSION TITLE
x-bind not work after x-for and template tag

### DISCUSSION THREAD

**Discussion by Sulaymon-dev** [alpinejs/alpine#1939] [7. Bugs (archived)] [answered]
I create re-usable dropdown component
when I use it before template tag it's work fine, but after template tag it's not work

https://codepen.io/sulaymon-dev/pen/xxdMZbW

**Comment by SimoTod** [comment, marked as answer]
I appreciate that this is not relevant to the bug but why do you set x-bind using vanilla javascript in the init function?
This example would work
```html
<div x-data="data()">
    <div x-data="useDropdown">
        <button x-bind="ref__dropdown">OK</button>
        <div x-bind="box__dropdown">box</div>
    </div>

    <table>
        <template x-for="d in data" :key="d.id">
            <tr>
                <td x-text="d.text"></td>
            </tr>
        </template>
    </table>
     
    <div x-data="useDropdown">
        <button x-bind="ref__dropdown">OK</button>
        <div x-bind="box__dropdown">box</div>
    </div>
</div>
```

I think x-for pauses the mutation observer and it probably has some setTimeout that allows the next component to start up at the same time, so the dynamically inserted x-bind doesn't get evaluated
